### PR TITLE
[PACT-649] Added 'administrative / biographical' field to record.

### PR DESCRIPTION
**Pull request by tna-erasmos**

### CODE REVIEW

**Comment by tna-erasmos** [comment]
You can view a demo [here](https://drive.google.com/file/d/17mrgEPgx_916mY-QYWRhIwBumyw1TvUG/view?usp=sharing).

**Review by felixgb — APPROVED** [review]

File: test/controllers/EditSetControllerSpec.scala [hunk 1]
```diff
@@ -1701,6 +1702,47 @@ class EditSetControllerSpec extends BaseSpec {
 
           }
         }
+        "background" when {
+          "is too long" in {
+
+            val excessivelyLongBackground = "Something about one of the people." * 400
+            val values = validValuesForSaving ++ Map("background" -> excessivelyLongBackground)
+
+            val result = submitWhileLoggedIn(1, "COAL.2022.V5RJW.P", values)
```
Comment (userjd) [inline test/controllers/EditSetControllerSpec.scala:1711]: @tna-erasmos for consistency this record id needs to match with the expected record id  `COAL.2022.V1RJW.P`

File: test/controllers/EditSetControllerSpec.scala [hunk 2]
```diff
@@ -1836,6 +1878,42 @@ class EditSetControllerSpec extends BaseSpec {
             )
 
           }
+          "the 'background' field is blank" in {
+
+            val values = validValuesForSaving ++ Map("background" -> "")
+
+            val editRecordPageResponse = submitWhileLoggedIn(1, "COAL.2022.V5RJW.P", values)
```
Comment (userjd) [inline test/controllers/EditSetControllerSpec.scala:1885]: same as above

File: test/controllers/EditSetControllerSpec.scala [hunk 3]
```diff
@@ -1836,6 +1878,42 @@ class EditSetControllerSpec extends BaseSpec {
...
+
+            status(editRecordPageResponse) mustBe SEE_OTHER
+            redirectLocation(editRecordPageResponse) mustBe Some("/edit-set/1/record/COAL.2022.V1RJW.P/edit/save")
+
+            val getRecordResult = getRecordForEditingWhileLoggedIn(1, "COAL.2022.V5RJW.P")
```
Comment (userjd) [inline test/controllers/EditSetControllerSpec.scala:1890]: same as above

File: test/controllers/EditSetControllerSpec.scala [hunk 4]
```diff
@@ -1836,6 +1878,42 @@ class EditSetControllerSpec extends BaseSpec {
...
+            assertPageAsExpected(
+              asDocument(getRecordResult),
+              ExpectedEditRecordPage(
+                title = "Edit record",
+                heading = "TNA reference: COAL 80/80/5",
+                legend = "Intellectual properties",
+                classicCatalogueRef = "COAL 80/80/5",
+                omegaCatalogueId = "COAL.2022.V5RJW.P",
```
Comment (userjd) [inline test/controllers/EditSetControllerSpec.scala:1898]: same as above

**Review by userjd — CHANGES_REQUESTED** [review]

**Comment by tna-erasmos** [comment]
@rwalpole [pointed out](https://tna-digital.slack.com/archives/CQB3ETSS3/p1673010600508979?thread_ts=1672925621.155229&cid=CQB3ETSS3) that the error message in the summary should link to the field; I'll fix that.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 5]
```diff
@@ -218,6 +218,19 @@
         )
     },
     attributes = Map("rows" -> "5", "autofocus" -> "true")))
+@govukTextarea(Textarea(
+    id = "background",
+    name = "background",
+    value = editSetRecordForm.data.get("background"),
+    label = Label(forAttr = Some("background"), content = Text(messages("edit-set.record.edit.background"))),
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:225]: The `Textarea` needs a `classes` field set to the string value "govuk-label--s" to include this an an additional class, as shown in the prototype. You can read more about this at https://design-system.service.gov.uk/get-started/labels-legends-headings/

**Review by rwalpole — CHANGES_REQUESTED** [review]

File: test/controllers/EditSetControllerSpec.scala (hunk 1 above)
Comment (tna-erasmos) [reply test/controllers/EditSetControllerSpec.scala]: Changed.

File: test/controllers/EditSetControllerSpec.scala (hunk 2 above)
Comment (tna-erasmos) [reply test/controllers/EditSetControllerSpec.scala:1962]: Changed.

File: test/controllers/EditSetControllerSpec.scala (hunk 3 above)
Comment (tna-erasmos) [reply test/controllers/EditSetControllerSpec.scala:1967]: Changed.

File: test/controllers/EditSetControllerSpec.scala (hunk 4 above)
Comment (tna-erasmos) [reply test/controllers/EditSetControllerSpec.scala:1975]: Changed.

**Comment by tna-erasmos** [comment]
About the bug where the summary error messages weren't linking correctly: I have fixed this for all of the fields, as it was almost easier than doing them separately (as such, this [other story](https://national-archives.atlassian.net/browse/PACT-651) can be closed or deleted).

[Here is a demo video](https://drive.google.com/file/d/1F5GsvgFOylJI5_wiURGIznW3eS4O2mTj/view?usp=sharing) of the fixed behaviour.

**Review by felixgb — APPROVED** [review]

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 5 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: image

Changed.

File: test/controllers/EditSetControllerSpec.scala (hunk 4 above)
Comment (userjd) [reply test/controllers/EditSetControllerSpec.scala:1993]: LGTM

File: test/controllers/EditSetControllerSpec.scala (hunk 3 above)
Comment (userjd) [reply test/controllers/EditSetControllerSpec.scala:1985]: LGTM

File: test/controllers/EditSetControllerSpec.scala (hunk 2 above)
Comment (userjd) [reply test/controllers/EditSetControllerSpec.scala:1980]: LGTM

**Review by userjd — APPROVED** [review]

File: test/controllers/EditSetControllerSpec.scala (hunk 1 above)
Comment (userjd) [reply test/controllers/EditSetControllerSpec.scala]: LGTM

**Comment by tna-erasmos** [comment]
Despite the two approvals, I think it's best that we ensure that @rwalpole is satisfied with the changes.

**Review by rwalpole — APPROVED** [review]

**Pull request merged**